tests/common/devices/sonic_asic.py

```diff
@@ -387,6 +340,53 @@ def run_redis_cmd(self, argv=[]):
 
         return result["stdout_lines"]
 
+    def port_exists(self, port):
+        """
+        Check if a given port exists in ASIC instance
+        Args:
+            port: port ID
+        Returns:
+            True or False
+        """
+        if self.ports is not None:
+            return port in self.ports
+
+        if_db = self.show_interface(
+            command="status"
+        )["ansible_facts"]["int_status"]
+
+        self.ports = set(if_db.keys())
+        return port in self.ports
+
+    def get_queue_oid(self, port, queue_num):
+        """
+        Get the queue OID of given port and queue number. The queue OID is
+        saved for the purpose of returning the ASIC instance of the
+        queue OID
+
+        Args:
+            port: Port ID
+            queue_num: Queue
+        Returns:
+            Queue OID
+        """
+        redis_cmd = [
+            "redis-cli", "-n", "2", "HGET", "COUNTERS_QUEUE_NAME_MAP",
+            "{}:{}".format(port, queue_num)
+        ]
+        queue_oid = next(iter(self.run_redis_cmd(redis_cmd)), None)
```

Comment:
Why do we need an iterator here?

Reply:
if command returns nothing, this gets assigned to None. "test" and "assign" in one statement.